module-2/backend/week_one.md

```diff
@@ -4,31 +4,33 @@ Fork this respository. Answer the questions to the best of your ability. Try to
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
-21. What's the difference between agile workflow and waterfall method?
-22. What is the difference between `#new` and `#create`?
+9. What's the purpose of ERB? erb allows us a way to interpolate ruby code in an HTML form, which is awesome because without this you can't run ruby code in HTML
+10. Why do I need a development AND test database? becuase you can REALLY screw up your data if you're pulling from the same files for both testing and development 
+11. What's responsive design? website design that scales for the device being used, a.k.a. it knows if you're viewing for mobile, desktop, etc. and automatically loads for whatever view is supported on that device
+12. What is CRUD and why is it important? it's a protocol for adding functionality to webpages (create, read, update, delete)
+13. What does HTTP stand for? hypertext transfer protocol
+14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways? you can do this (<%=... %>), which will show on the page whatever the result of the code being run is; or you can do this (<% ... %>), which does NOT show what the result of the code being run is, but still runs the code
+15. What's an ORM? stands for Object Relational Mapping and is used to allow us to use Ruby code (or Python or some other OO language) to interact with database lanagues
+16. What's the most commonly used ORM? ActiveRecord (for Ruby)
+17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for. OK, you will need to have GET '/restaurants' to see/read all restaurants, you'll want GET 'restaurants/new' to add new places, you'll need POST '/restuarants' to add the new place, you'll want GET '/restaurants/:id' to show just one, you'll want GET '/restaurants/:id/edit' to edit info for one at a time, you'll want, you'll want PUT '/restaurant/:id' to update the changes, and you'll want DELETE '/restaurants/:id' to delete 
+18. What's a migration? it sends a table into a database and sets the table up for entries 
+19. When you create a migration, does it automatically modify your database? If you have changed the data you want in the table, each migration will change the database, if not, then no. 

```

Comment:
I think the key word here is *automatically*. When you create a migration it won't make a change to the DB until you run `rake db:migrate`.